lib/api/api_auth_test.go

```diff
@@ -65,3 +65,53 @@ func TestAuthLDAPSendsCorrectBindDNWithNoTemplate(t *testing.T) {
 		t.Fatalf("ldapTemplateBindDN should be %s != %s", expectedDn, templatedDn)
 	}
 }
+
+func TestNoAuthPathsIsImmutable(t *testing.T) {
+	t.Parallel()
+
+	original := noAuthPaths()
+
+	mutated := noAuthPaths()
+	for i := range mutated {
+		mutated[i] = mutated[i] + "flrglgrgldrgl"
+	}
+
+	final := noAuthPaths()
+
+	if len(final) != len(original) {
+		t.Fatal("noAuthPaths is not immutable")
+	}
+	for i := range original {
+		if final[i] != original[i] {
+			t.Fatal("noAuthPaths is not immutable")
+		}
+		if final[i] == mutated[i] {
+			t.Fatal("noAuthPrefixes is not immutable")
+		}
+	}
+}
+
+func TestNoAuthPrefixesIsImmutable(t *testing.T) {
+	t.Parallel()
+
+	original := noAuthPrefixes()
+
+	mutated := noAuthPrefixes()
+	for i := range mutated {
+		mutated[i] = mutated[i] + "flrglgrgldrgl"
+	}
+
+	final := noAuthPrefixes()
+
+	if len(final) != len(original) {
+		t.Fatal("noAuthPrefixes is not immutable")
+	}
+	for i := range original {
+		if final[i] != original[i] {
+			t.Fatal("noAuthPrefixes is not immutable")
+		}
+		if final[i] == mutated[i] {
+			t.Fatal("noAuthPrefixes is not immutable")
```

Comment:
My main thinking was that this test should catch if in the future one were to "optimize" these functions by replacing them with module variables instead. Since Go doesn't have immutable slice constants, such a refactoring would create an opportunity to corrupt these exception lists and possibly circumvent the authorization layer. It's an important security assumption that these exception lists are immutable, so that assumption should have a test to verify it.

But yes, a refactor to return a pointer to a long-lived mutable object would have the same issue, though I hadn't thought of that particular case.